xtask/src/semver_check.rs

```diff
@@ -0,0 +1,294 @@
+    chips: Vec<Chip>,
+) -> Result<(), anyhow::Error> {
+    for package in packages {
+        log::info!("Generating API baseline for {package}");
+
+        for chip in &chips {
+            log::info!("Chip = {}", chip.to_string());
+            let package_name = package.to_string();
+            let package_path = crate::windows_safe_path(&workspace.join(&package_name));
+
+            let current_path = build_doc_json(package, chip, &package_path)?;
+
+            remove_unstable_items(&current_path)?;
+
+            let file_name = if package.chip_features_matter() {
+                format!("{}", chip.to_string())
+            } else {
+                "api".to_string()
+            };
+
+            let to_path = PathBuf::from(&package_path).join(format!("api-baseline/{}.json.gz", file_name));
+            fs::create_dir_all(to_path.parent().unwrap())?;
+
+            log::debug!("Compress into {current_path:?}");
+            let mut encoder = flate2::write::GzEncoder::new(fs::File::create(to_path)?, flate2::Compression::default());
+            encoder.write_all(&std::fs::read(current_path)?)?;
+
+            if !package.chip_features_matter() {
+                break;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+pub fn check(
+    workspace: &PathBuf,
+    packages: Vec<Package>,
+    chips: Vec<Chip>,
+) -> Result<(), anyhow::Error> {
+    for package in packages {
+        log::info!("Semver-check API for {package}");
+
+        for chip in &chips {
+            log::info!("Chip = {}", chip.to_string());
+            let package_name = package.to_string();
+            let package_path = crate::windows_safe_path(&workspace.join(&package_name));
+
+            let current_path = build_doc_json(package, chip, &package_path)?;
+            remove_unstable_items(&current_path)?;
+
+            let file_name = if package.chip_features_matter() {
+                format!("{}", chip.to_string())
+            } else {
+                "api".to_string()
+            };
+
+            let baseline_path_gz = PathBuf::from(&package_path).join(format!("api-baseline/{}.json.gz", file_name));
+
+            let baseline_path = temp_file::TempFile::new()?;
+            let buffer = Vec::new();
+            let mut decoder = flate2::write::GzDecoder::new(buffer);
+            decoder.write_all(&(fs::read(&baseline_path_gz)?))?;
+            fs::write( baseline_path.path(), decoder.finish()?)?;
+
+            let mut check = cargo_semver_checks::Check::new(
+                cargo_semver_checks::Rustdoc::from_path(current_path),
+            );
+            check.set_baseline(cargo_semver_checks::Rustdoc::from_path(baseline_path.path()));
+            let mut cfg = cargo_semver_checks::GlobalConfig::new();
+            // seems there is no way of getting the details via the API - so lets log them
+            cfg.set_log_level(Some(log::Level::Info));
+            let check_result = check.check_release(&mut cfg)?;
```

Comment:
Okay this is a bit hard to read, after a few lines the word `check` loses its meaning. But since we're blocking this PR on the compiler, could you tweak this a bit so that we'll have a `minimum_update` function that we can call for each package? I'd love to use this as part of the version bump command